### PR TITLE
mix: update to amqp 3.0 and lift elixir requirement to 1.11, bump all deps

### PR DESCRIPTION
**Pull request by dch**
## Description

After many years of pain and grief we can finally get rid of lager and the noisy logging in a clean fashion. I've been running with amqp 3.0 on OTP24 in a simple production setup without any issues so far. Fantastic!

What's the general deprecation policy?

The upstream OTP team only provides support for Erlang >= OTP22 at least, which would imply Elixir 1.11 as oldest version, and pma's AMQP only supports 23+.  rabbittmq_common only supports OTP22.1 up.

## Lager

- https://ferd.ca/erlang-otp-21-s-new-logger.html
- https://github.com/pma/amqp/wiki/Versions-and-Compatibilities#3x
- https://github.com/pma/amqp/wiki/3.0-Release-Notes#breaking-changes-and-upgrade-guide
- https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.9.0

## Tests

All against RabbitMQ 3.9.5 (almost latest, 3.9.6 just out today)

- OTP24 + Elixir 1.12.3 100% no output
- OTP23 + Elixir 1.11.3 100% see below
- OTP22 + Elixir 1.11.3 100% see below
- OTP21 - fails to compile (hard requirement from mimerl via hackney via excoveralls)
- removing that still hits credentials_obfuscation with `OTP release 22.1 or later is required. Version in use: 21.3.8.24`

I'm not yet sure how best to enforce this, we could add some checks using https://hexdocs.pm/elixir/System.html#version/0 & https://hexdocs.pm/elixir/System.html#otp_release/0 which are available since ~ Elixir 1.9 at least, not sure if they should go in mix.exs or config.exs. The former makes more sense to me.

## OTP23 logs

```elixir
Generated gen_rmq app
.....................................
07:59:04.974 [error] GenServer #PID<0.2074.0> terminating
** (stop) :connection_closed
Last message: {:EXIT, #PID<0.2056.0>, :connection_closed}
............................................................................

Finished in 20.7 seconds
113 tests, 0 failures
```

## OTP22 logs

````elixir
Generated gen_rmq app
.......................
07:55:48.018 [error] GenServer #PID<0.1832.0> terminating
** (stop) :cancelled
Last message: {:EXIT, #PID<0.1814.0>, :cancelled}
..........................
07:55:54.536 [error] GenServer #PID<0.3692.0> terminating
** (stop) :cancelled
Last message: {:EXIT, #PID<0.3674.0>, :cancelled}
...............
07:55:58.068 [error] GenServer #PID<0.4160.0> terminating
** (stop) :cancelled
Last message: {:EXIT, #PID<0.4142.0>, :cancelled}
.................................................

Finished in 20.0 seconds
113 tests, 0 failures
```

### CODE REVIEW

**Comment by vorce** [comment]
Wow thanks @dch ! I'll have a look

**Comment by vorce** [comment]
@dch Can you also update .github/workflows/main.yml maybe something like:

static_analysis:
```yaml
...
- name: Set up Elixir
        uses: erlef/setup-beam@v1
        with:
          elixir-version: '1.11'
          otp-version: '22.3'
...
```

unit_test:
```yaml
...
strategy:
      matrix:
        elixir:
          - '1.11'
          - '1.12'
        otp:
          - '22.3'
          - '23.3.4.7'
          - '24.0.5'
...
```

File: mix.exs [hunk 1]
```diff
@@ -1,13 +1,13 @@
 defmodule GenRMQ.Mixfile do
   use Mix.Project
 
-  @version "3.0.0"
+  @version "3.0.1"
```
Comment (vorce) [inline mix.exs:4]:
```suggestion
  @version "4.0.0"
```
I think I want to actually bump the major version here. What do you think @mkorszun ?

Comment (dch) [reply mix.exs:4]: I agree, the lager / amqp / rabbitmq_common changes should drive this

**Comment by dch** [comment]
@vorce can you kick CI on this? I use the `...x` tags in the actions file, according to https://github.com/erlef/setup-beam this should just do the versions per https://erlang.org/download/ and not need version bumping in future

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -73,12 +73,12 @@ jobs:
...
-          - '1.9'
-          - '1.10'
+          - '1.11'
+          - '1.12'
         otp:
-          - '22.0'
-          - '23.0'
+          - '22.x'
+          - '23.x'
+          - '24.x'
```
Comment (vorce) [inline .github/workflows/main.yml:81]: Nice!

**Review by vorce — APPROVED** [review]
This looks great 🙌

**Comment by vorce** [comment]
Please take a look when you have time @mkorszun

**Comment by dch** [comment]
@mkorszun @akoutmos any thoughts?

**Review by mkorszun — APPROVED** [review]
Nice 🏅

**Comment by vorce** [comment]
Awesome, thanks again @dch 🙌

**Pull request merged**